### ISSUE TITLE
Clarify first 2d game HUD connection to Main

### ISSUE THREAD

**Issue by ido-trumer** [godotengine/godot-docs#8449]
**Your Godot version:**
4.1.3 stable

**Issue description:**
The instructions in [here](https://docs.godotengine.org/en/stable/getting_started/first_2d_game/06.heads_up_display.html#connecting-hud-to-main) are a bit confusing - they do not specify that you need to choose the HUD node, and not main.

This is a common issue, check out the following discussion - https://ask.godotengine.org/34238/how-to-connect-a-signal-to-a-function (check the 10 upvote comment)

This is the paragraph I suggest changing:

In the Node tab, connect the HUD's start_game signal to the new_game() function of the Main node by clicking the "Pick" button in the "Connect a Signal" window and selecting the new_game() method or type "new_game" below "Receiver Method" in the window. Verify that the green connection icon now appears next to func new_game() in the script.

I suggest changing it to:

In the HUD's Node tab, connect the ``start_game`` signal to the ``new_game()``
function of the Main node. Right Click the ``start_game`` signale, choose "Connect", 
and choose the "Pick" button in the "Connect a Signal" window. Select the ``new_game()`` 
method or type "new_game" below "Receiver Method" in the window.  
Verify that the green connection icon now appears next to ``func new_game()`` in the script.

This makes it clearer that we are talking about the HUD node, as well as a bit more elaborate on how to access all the menu items.

**URL to the documentation page (if already existing):**
https://docs.godotengine.org/en/stable/getting_started/first_2d_game/06.heads_up_display.html#connecting-hud-to-main

note - I tried to create a PR after reading these [instructions](https://docs.godotengine.org/en/latest/contributing/documentation/contributing_to_the_documentation.html#editing-pages-online), but I didn't have the option to "Fork this project and edit the file", so I've opened this issue.

**Comment by ndelah** [comment]
As a beginner your explanation is clearer than the documentation. However it's not possible to 'pick' the `new_game()` method. 

As you can see from the screenshot below, the connect** to a signal method** menu is opened from the **start_game()** signal located in the **hud** node. After clicking `pick`  the **select method** menu is opened. From that menu it should be possible to choose the `new_game()` method, however it is absent.  

As a beginner it's unclear what the next steps are or whether I committed a mistake. 
 
<img width="1407" alt="Screenshot 2023-12-21 at 16 39 33" src="https://github.com/godotengine/godot-docs/assets/18349236/f9465d5b-c622-455a-8273-403e6817ccb1">

I successfully implemented all the previous steps

**Comment by ndelah** [comment]
I would reword ido-trumer's suggestion as follows:
From the main scene, select the HUD node you just instantiated. On the right hand-side under the Node tab right click the `start_game()` signal and click **connect**.
Choo the "Pick" button in the "Connect a Signal" window. Select the new_game()
method or type "new_game" below "Receiver Method" in the window.
Verify that the green connection icon now appears next to func new_game() in the script.

**Comment by TesFalcon** [comment]
I'm just getting started in Godot. I have version 4.2.1. There are a LOT of assumptions being made in this 1st 2D game tutorial. I am LOST.
I went through the player portion, and it was detailed enough that I felt like I understood what was happening. It all worked with no issues.
At the very top of the mob portion (why is it called a mob not squid or enemy?), it seems like they start leaving out steps and LOTS of explanations.
New Scene? Ok, done.

RigidBody2D & Child Nodes? Ok, I did that, but I'm not seeing the button to deselect the children like I saw with the player. Where's that at?

Under CollisionObject2D node in the Inspector, there is NO Collision group.
If I search for "mask", I see a mask item as it shows in the picture under "CanvasItem" > "Visibility" which shows (among other things) Light Mask & Visibility Layer. 
I FOUND IT! As I was trying to describe this more clearly to you, I saw that the instructions weren't stepping down the tree. It's talking about stuff interchangably on either side of the screen and calling things by their old names rather than using the new name. All these name are VERY similar to each other. Instead, this is supposed to be under the "MOB" node that it keeps calling "RigidBody2D" even though we already changed the name of it. I thought this was under the CollisionShape2D node since it kept talking about the CollisionObject2D section, but they never define what a section is. UGH! 

This whole struggle happened again when setting up the collision area. Rather than giving top-down instructions, Transform > Rotation: Set to 90*. It jumps down to the property that needs changing (Rotation) and sets the heading where I'd find it in a parenthetical. (Does anyone even read parenthetical statements?)

My code is duplicated, but it all saves as the same player.gd script. 
It says add script to the enemy/mob thing of "extends RigidBody2D". 
Do I need to attach a new enemy script instead of using the same player script?
When it says to add x code to the _ready() function is that in addition to or to replace the code that's already there?

I got particularly frustrated when it came to the last part which states:
"The last piece is to make the mobs delete themselves when they leave the screen. Connect the screen_exited() signal of the VisibleOnScreenNotifier2D node to the Mob and add this code:
GDScript
func _on_visible_on_screen_notifier_2d_screen_exited():
	queue_free()"

I have tried the only 2 ways I know to accomplish this, but it's not working.
When I go to "VisibleOnScreenNotifier2D" to add the signal, it keeps asking me for a valid method. (???)
If I add the GDScript to the mash of code, I can't connect it to the signal. 
Right now this signal is just a huge flashing red light with klaxons.
I'm stumped, stuck, and frustrated.

**Comment by skyace65** [comment]
The documentation literally states "connect the HUD's start_game signal" that means ``start_game`` is the HUD's signal. I don't know why you would think to check the Main node for the HUD node's signal.

Ndeleh, your issue is you're inside of the HUD scene and not the main scene. It's not going to give you the option to connect to the Main node at all because it doesn't exist in that scene.

**Comment by JoshStLee** [comment]
I'm a beginner on Godot and I'm suggesting another clarification might be needed on C# users are required to build the HUD before the suggested text, instead of being commented into the code where not many users will notice it right away

> C# users are required to rebuild the project so the editor knows about the new signal.
>In the HUD's Node tab, connect the `start_game` signal to the `new_game()` function of the Main node. Right Click the `start_game` signale, choose "Connect", and choose the "Pick" button in the "Connect a Signal" window. Select the `new_game()` method or type "new_game" below "Receiver Method" in the window. Verify that the green connection icon now appears next to `func new_game()` in the script.
